WordPress/Classes/ViewRelated/Notifications/Tools/KeyboardDismissHelper.swift

```diff
@@ -0,0 +1,191 @@
+        nc.addObserver(self, selector: #selector(keyboardDidHide), name: UIKeyboardDidHideNotification, object: nil)
+
+    }
+
+    /// Removes all of the Keyboard Event Listeners
+    ///
+    func stopListeningToKeyboardNotifications() {
+        NSNotificationCenter.defaultCenter().removeObserver(self)
+    }
+
+
+
+    /// ScrollView willBeginDragging Event
+    ///
+    func scrollViewWillBeginDragging(scrollView: UIScrollView) {
+        guard let dismissableControl = dismissableControl where isKeyboardVisible == true else {
+            return
+        }
+
+        initialControlPositionY = dismissableControl.frame.maxY
+        initialBottomConstraint = bottomLayoutConstraint.constant
+        trackingDragOperation = true
+    }
+
+    /// ScrollView didScroll Event
+    ///
+    func scrollViewDidScroll(scrollView: UIScrollView) {
+        guard trackingDragOperation == true else {
+            return
+        }
+
+        let location = scrollView.panGestureRecognizer.locationInView(parentView)
+        let delta = location.y - initialControlPositionY
+        let newConstant = min(max(initialBottomConstraint - delta, 0), initialBottomConstraint)
+
+        guard newConstant != bottomLayoutConstraint.constant else {
+            return
+        }
+
+        bottomLayoutConstraint.constant = newConstant
+        parentView.layoutIfNeeded()
+    }
+
+    func scrollViewWillEndDragging(scrollView: UIScrollView, withVelocity velocity: CGPoint) {
+        trackingDragOperation = false
+    }
+
+
+    // MARK: - Notification Helpers
+    func keyboardWillShow(note: NSNotification) {
+        isKeyboardVisible = true
+        handleKeyboardFrameChange(note)
+        onWillShow?()
+    }
+
+    func keyboardDidShow(note: NSNotification) {
+        handleKeyboardFrameChange(note)
+        onDidShow?()
+    }
+
+    func keyboardWillHide(note: NSNotification) {
+        isKeyboardVisible = false
+        handleKeyboardFrameChange(note)
+        onWillHide?()
+    }
+
+    func keyboardDidHide(note: NSNotification) {
+        handleKeyboardFrameChange(note)
+        onDidHide?()
+    }
+
+
+    // MARK: - Private Helpers
+
+    private func handleKeyboardFrameChange(note: NSNotification) {
+        guard let kbRect = note.userInfo?[UIKeyboardFrameEndUserInfoKey] as? NSValue,
+            let duration = note.userInfo?[UIKeyboardAnimationDurationUserInfoKey] as? NSTimeInterval,
+            let rawCurve = note.userInfo?[UIKeyboardAnimationCurveUserInfoKey] as? Int,
+            let curve = UIViewAnimationCurve(rawValue: rawCurve) else
+        {
+            return
+        }
+
+        // Bottom Inset: Consider the tab bar!
+        let convertedKeyboardRect = parentView.convertRect(kbRect.CGRectValue(), fromView: nil)
+        let bottomInset = max(convertedKeyboardRect.height - convertedKeyboardRect.maxY + parentView.frame.height, 0)
+
+        UIView.beginAnimations(nil, context: nil)
```

Comment:
Reason why is... the `animationCurve` parameter. Although the blocks API accepts the `UIViewAnimationOptions`, and the Curve can be converted... problem there si that there's a `delay` parameter.

Apparently, setting the delay to Zero still causes the animation to be fired with an *extremely small* delay, noticeable with Slow Animations enabled.

Furthermore, calling the `setAnimationCurve` function before running the plain block-based API seems to be disregarding the (statically-set) curve. I'm afraid that... the old school API is what brings us closer to the actual Keyboard animation.

(**Just** tested on iOS 9 simulator with Slow Animations enabled).

Thanks for pointing it out!!!